java/client/src/main/java/redis/clients/jedis/Jedis.java

```diff
@@ -6054,4 +6060,1436 @@ public int getDB() {
+        ensureInitialized();
+        try {
+            return glideClient.lpos(key, element).get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new JedisException("LPOS operation failed", e);
+        }
+    }
+
+    /**
+     * Returns the index of the first matching element in the list stored at key (binary version).
+     *
+     * @param key the key of the list
+     * @param element the element to search for
+     * @return the index of the first matching element, or null if not found
+     */
+    public Long lpos(final byte[] key, final byte[] element) {
+        checkNotClosed();
+        ensureInitialized();
+        try {
+            return glideClient.lpos(GlideString.of(key), GlideString.of(element)).get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new JedisException("LPOS operation failed", e);
+        }
+    }
+
+    /**
+     * Returns the index of matching elements in the list stored at key with additional options.
+     *
+     * @param key the key of the list
+     * @param element the element to search for
+     * @param params additional parameters for the search
+     * @return the index of the matching element, or null if not found
+     */
+    public Long lpos(String key, String element, LPosParams params) {
+        checkNotClosed();
+        ensureInitialized();
+        try {
+            LPosOptions options = LPosOptions.builder().build();
+            if (params.getRank() != null) {
+                options = LPosOptions.builder().rank(((long) params.getRank())).build();
+            }
+            if (params.getMaxlen() != null) {
+                options = LPosOptions.builder().maxLength(((long) params.getMaxlen())).build();
+            }
+            if (params.getRank() != null && params.getMaxlen() != null) {
+                options =
+                        LPosOptions.builder()
+                                .rank(((long) params.getRank()))
+                                .maxLength(((long) params.getMaxlen()))
+                                .build();
+            }
+            return glideClient.lpos(key, element, options).get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new JedisException("LPOS operation failed", e);
+        }
+    }
+
+    /**
+     * Returns the index of matching elements in the list stored at key with additional options
+     * (binary version).
+     *
+     * @param key the key of the list
+     * @param element the element to search for
+     * @param params additional parameters for the search
+     * @return the index of the matching element, or null if not found
+     */
+    public Long lpos(final byte[] key, final byte[] element, LPosParams params) {
+        checkNotClosed();
+        ensureInitialized();
+        try {
+            LPosOptions options = LPosOptions.builder().build();
+            if (params.getRank() != null) {
```

Comment:
This is weird. Why do we check getRank(), create a builder, then check getMaxlen() and create a builder, then check both then create a builder?

Why not create a builder, check rank, update the builder, then check maxLen, then update the builder?

Reply:
Can you check other methods using Params objects to see if that redundancy is also there?